### DISCUSSION TITLE
Default name position

### DISCUSSION THREAD

**Discussion by rijkvanmanen** [5axes/NameIt#10] [Ideas]
I have a feature request for this awesome plugin: add an option to "Define Text Parameters" to place the text mirrored in the center of the print. In this way only the line pattern of the first layer is modified, without adding to much print time.
![image](https://user-images.githubusercontent.com/37628975/176691995-2d6ff7a2-4858-450d-a36b-a4bbe23a387f.png)

**Comment by 5axes** [comment]
Nice improvement. It's not really difficult to integrate but : 

- The "Center of the print" is not necessary  a print area  Ie: If we have a cylinder, a U  or something like that.
- The length of the text could be larger than the size of the part

**Reply by rijkvanmanen** [reply]
Sure, it will not always work. That is also why I suggest to make it an option. Maybe there are even more logical default locations, but I think you covered most scenarios' if you place the text in front of the object (like it is now) or as I suggest.

**Reply by 5axes** [reply]
I'm a little hesitant about what to name the option: 
Middle Mode ?
![image](https://user-images.githubusercontent.com/11015345/176717840-55d47197-aa08-41f4-b7c1-5bef56b6beeb.png)

**Reply by rijkvanmanen** [reply]
naming is also not my expertise :)
But what about making it a drop down menu with (currently) two options: "front", "center bottom"?

**Comment by 5axes** [comment]
[V 1.5.0 Updated](https://github.com/5axes/NameIt/releases/tag/V1.5.0)
![image](https://user-images.githubusercontent.com/11015345/176722496-e2f8d5f9-e2ef-4631-90ca-a07c31960899.png)

**Comment by 5axes** [comment]
Possible improvement : Fix for the text mesh Infill 0% Nb Wall=0 
![image](https://user-images.githubusercontent.com/11015345/176816035-fb5fa338-f3af-477a-842b-0af63ac04f40.png)

Other advantage : If the text is a bigger than the part doesn't create  a geometry in this case.
![image](https://user-images.githubusercontent.com/11015345/176816824-60d23b7a-b524-4568-a25d-bff54be5dd20.png)

**Reply by rijkvanmanen** [reply]
nice update! But it does leave a gap in your first layer, in most cases that will be fine I think.

**Reply by 5axes** [reply]
That's the goal something like the well known 3D benchy .
![image](https://user-images.githubusercontent.com/11015345/177123263-c9ae8707-b915-4db4-88da-3317844ab018.png)

If you don't leave a gap it could be difficult according to your bed leveling , to read the text ?

**Comment by 5axes** [comment]
Could be also integrated as an option : Middle Filled Text Option  Need to have feedback from user on this point.

![image](https://user-images.githubusercontent.com/11015345/177136339-bb61857f-f05d-424b-a989-31d22f17a822.png)

Option Checked : 
![image](https://user-images.githubusercontent.com/11015345/177136605-cb7aacb6-654a-43b5-819b-caecf298c327.png)

Option not Checked (Default) : 
![image](https://user-images.githubusercontent.com/11015345/177136787-1ce862c7-d615-4e5a-a8c7-3c5689b749e3.png)

**Reply by rijkvanmanen** [reply]
Adding it as an option does make sense. And add it to a drop down to keep it clean. With the dropdown items: ['In front of print', 'bottom center', 'bottom center (not filled)']
![image](https://user-images.githubusercontent.com/37628975/177140824-8dade756-4bed-44c8-a57a-26e886a25120.png)

**Reply by 5axes** [reply]
Next release .. Yes nice idea !

**Reply by 5axes** [reply]
V1.6.1
